Mathlib/CategoryTheory/Limits/FintypeCat.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2023 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.CategoryTheory.Limits.Shapes.FiniteLimits
+import Mathlib.CategoryTheory.Limits.Types
+import Mathlib.CategoryTheory.Limits.Creates
+import Mathlib.CategoryTheory.Limits.Preserves.Finite
+import Mathlib.Data.FunLike.Fintype
+
+/-!
+# (Co)limits in the category of finite types
+
+We show that finite (co)limits exist in `FintypeCat` and that they are preserved by the natural
+inclusion `FintypeCat.incl`.
+-/
+
+open CategoryTheory Limits Functor
+
+universe u
+
+namespace CategoryTheory.Limits.FintypeCat
+
+/-- Any functor from a finite category to Types that only involves finite objects,
+has a finite limit. -/
+noncomputable def finiteLimitOfFiniteDiagram {J : Type} [SmallCategory J] [FinCategory J]
+    (K : J ⥤ Type*) (_ : (j : J) → Finite (K.obj j)) : Fintype (limit K) := by
+  have : Fintype (sections K) := Fintype.ofFinite ↑(sections K)
+  exact Fintype.ofEquiv (sections K) (Types.limitEquivSections K).symm
+
+noncomputable instance inclusionCreatesFiniteLimits {J : Type} [SmallCategory J] [FinCategory J] :
+    CreatesLimitsOfShape J FintypeCat.incl.{u} where
+  CreatesLimit := by
+    intro K
+    have : Fintype (limit (K ⋙ FintypeCat.incl.{u})) := by
+      apply finiteLimitOfFiniteDiagram (K ⋙ FintypeCat.incl)
+      intro j
+      simp only [comp_obj, FintypeCat.incl_obj]
+      exact inferInstance
+    exact createsLimitOfFullyFaithfulOfIso (FintypeCat.of <| limit <| K ⋙ FintypeCat.incl)
+      (eqToIso rfl)
```

Comment:
```suggestion
      (Iso.refl _)
```
If does not really matter here, but in general, it is advisable to use `Iso.refl _` if this is possible.

Reply:
Changed in the latest commit.